### ISSUE TITLE
Please tell me about how to remove the p tag wrapping

### ISSUE THREAD

**Issue by oceanfish567** [tinymce/tinymce#9601]
## 📝 Provide detailed reproduction steps (if any)
Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](https://fiddle.tiny.cloud) or similar.
1. …type <div itemprop="caption" class="image__caption attribution"><span data-editable="metaCaption" class="inline-placeholder">A view of Diamondhead on the Oahu, Hawaii, coastline. Downtown Honolulu is to the right of Diamondhead.</span></div>
2. …save it in source code window

### ✔️ Expected result
*What is the expected result of the above steps?*
If I enter html tag syntax in the editor, is there a way to prevent the editor from having p tag packaging? I hope it can be kept as it is?

### ❌ Actual result
*What is the actual result of the above steps?*
it will becaome: <p><div itemprop="caption" class="image__caption attribution"><span data-editable="metaCaption" class="inline-placeholder">A view of Diamondhead on the Oahu, Hawaii, coastline. Downtown Honolulu is to the right of Diamondhead.</span></div></p>
### ❓ Possible solution
*If you have ideas, you can list them here. Otherwise, you can delete this section.*
1. I saw someone on the Internet saying that forced_root_block: false and html_verify: false have not changed after I used them. Is it not supported by tinymce7?
2. A friend told me to type directly in the source code window. This method works.
## 📃 Other details
- Browser: …
- OS: …
- First affected version: …
- Worked in version: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by spocke** [comment]
I wasn't able to reproduce this if I paste that into the code editor then it's retained as is. TinyMCE follows HTML standards so DIV is not a valid child of P so never seen this before. So need a proper reproduction case here.

**Comment by oceanfish567** [comment]
> I wasn't able to reproduce this if I paste that into the code editor then it's retained as is. TinyMCE follows HTML standards so DIV is not a valid child of P so never seen this before. So need a proper reproduction case here.

yes, as long as I paste the html code in the source code, there will be no p packaging problem. tks.

**Comment by matthewjdewhurst-fw** [comment]
This has become a problem in our use case with Magento. We've recently updated to 2.4.6-p8, which bundled in an upgrade to TinyMCE 6. We recently decided, as part of our move to the Hyva theme, to disable the Page Builder to further improve performance and improve code maintainability with a widget/component based CMS approach.

So, using Magento's widget plugin for TinyMCE, our Marketing team can build pages up from components (widgets) developed and maintained by us. Previously, setting `forced_root_block` to empty did the trick for us, but no longer works as this usage is now [no longer supported](https://www.tiny.cloud/docs/tinymce/6/changelog/#removed).

In our use case, the contents of the TinyMCE is placed into a grid container on the frontend which handles the content flow and spacing (utilising grid gap). The addition of paragraph tags is now causing spacing issues with that grid container (large blank areas of whitespace).

I understand TinyMCE's point that using it this way may cause problems, but for those of us using TinyMCE in what might be considered the improper way and who don't need TinyMCE to work in the traditional way, a method of disabling this default behaviour would still be very useful.

**Comment by ShibenDuttaTR** [comment]
I am facing the exact same issue.
The version of tinyMCE is 5
Using formik to manage forms makes the form dirty as soon as tinyMCE adds the <p> tags
We need a way to avoid the <p> tags

**Comment by matthewjdewhurst-fw** [comment]
> I am facing the exact same issue. The version of tinyMCE is 5 Using formik to manage forms makes the form dirty as soon as tinyMCE adds the `<p>` tags We need a way to avoid the `<p>` tags

This isn't ideal, but we hot fixed the problem by adding the following JavaScript code to all our pages:

```javascript
var allParas = document.getElementsByTagName('p');
for(var i = 0; i < allParas.length; i++){
    if(allParas[i].innerHTML.trim().length === 0){
        allParas[i].style.display = 'none';
    }
}
```

Hides any paragraphs that are completely empty of content, like the paragraphs TinyMCE is adding.

You could use `allParas[i].remove` instead if you didn't want to just hide them.
